### ISSUE TITLE
Breaking change about path

### ISSUE THREAD

**Issue by sanemat** [packsaddle/ruby-ssh_url#14]
v1 behavior: #parse always returns path which start with '/', and Generic#to_s always returns path's first '/' stripped.
v2 behavior: #parse returns path which depends on input (absolute path or relative path)

Some people (include me) write code which depends on v1 behavior.

https://github.com/packsaddle/ruby-saddler-reporter-support-git

``` ruby
slug_regex = %r{\A/(?<slug>.*?)(?:\.git)?\Z}
remote_urls.map do |url|
  uri = GitCloneUrl.parse(url)
  match = slug_regex.match(uri.path)
  match[:slug] if match
end.compact.first
```

https://github.com/packsaddle/ruby-saddler-reporter-support-git/blob/acbf76444beb5fba6a8efa8c2fd3e5aabf3f027f/lib/saddler/reporter/support/git/repository.rb#L25-L30

https://github.com/robert2d/pully

``` ruby
def process_remote(remote)
  URI.parse(remote)
  return remote
rescue URI::InvalidURIError
  uri = URI::SshGit.parse(remote)
  return "https://#{uri.host}#{uri.path}"
end
```

https://github.com/robert2d/pully/blob/dabcc57cfb4df596f6a35928c949f55e79c88968/lib/pully.rb#L33-L39

**Comment by sanemat** [comment]
https://rubygems.org/gems/git_clone_url v1.0.1 is published.
git_clone_url v1.x uses uri-ssh_git v1.x.

**Comment by sanemat** [comment]
I recommend this workaround:

``` diff
-  spec.add_runtime_dependency 'uri-ssh_git'
+  spec.add_runtime_dependency 'uri-ssh_git', '>= 1.0', '< 2.0'
```

**Comment by sanemat** [comment]
I released `uri-ssh_git 2.0.0.pre.1`.
:frog:

**Comment by sanemat** [comment]
I released `git_clone_url 2.0.0.pre.1`. :football:

**Comment by sanemat** [comment]
I released `uri-ssh_git 2.0.0`. :taxi:
